extensions/scarb-snforge-test-collector/src/main.rs

```diff
@@ -0,0 +1,88 @@
+use anyhow::{Context, Result};
+use clap::Parser;
+use fs4::FileExt;
+use std::fs::{create_dir_all, File};
+use std::io::Write;
+
+use scarb_metadata::MetadataCommand;
+use scarb_ui::args::PackagesFilter;
+
+use crate::compilation::compile_tests;
+use crate::crate_collection::collect_test_compilation_targets;
+use crate::metadata::{
+    compilation_unit_for_package, corelib_for_compilation_unit, crates_config_for_compilation_unit,
+    dependencies_for_compilation_unit,
+};
+
+mod compilation;
+mod crate_collection;
+mod metadata;
+
+/// Starknet Foundry private extension for compiling test artifacts.
+/// Users should not call it directly.
+#[derive(Parser)]
+#[command(version)]
+struct Args {
+    #[command(flatten)]
+    packages_filter: PackagesFilter,
+}
+
+fn main() -> Result<()> {
+    let args = Args::parse();
+
+    let metadata = MetadataCommand::new().inherit_stderr().exec()?;
+    let selected_packages_metadata = args.packages_filter.match_many(&metadata)?;
+
+    let target_dir = metadata
+        .target_dir
+        .clone()
+        .unwrap_or_else(|| metadata.workspace.root.join("target"));
+    let snforge_target_dir = target_dir.join(&metadata.current_profile).join("snforge");
+    create_dir_all(&snforge_target_dir)?;
+
+    for package_metadata in selected_packages_metadata {
+        let compilation_unit_metadata = compilation_unit_for_package(&metadata, &package_metadata)?;
+
+        let compilation_targets = collect_test_compilation_targets(
+            &package_metadata.name,
+            &package_metadata.root,
+            compilation_unit_metadata.target.source_root(),
+            &compilation_unit_metadata.target.source_path,
+        )?;
+
+        let dependencies = dependencies_for_compilation_unit(compilation_unit_metadata);
+        let corelib_path = corelib_for_compilation_unit(compilation_unit_metadata)?;
+        let crates_config =
+            crates_config_for_compilation_unit(&metadata, compilation_unit_metadata);
+
+        let test_crates = compile_tests(
+            &compilation_targets,
+            &dependencies,
+            &corelib_path,
+            &crates_config,
+        )?;
+
+        let serialized_crates = serde_json::to_string(&test_crates)?;
+
+        // artifact saved to `{target_dir}/{profile_name}/{package_name}.sierra.json`
+        let output_path =
+            snforge_target_dir.join(&format!("{}.sierra.json", package_metadata.name));
```

Comment:
Maybe it would make sense to use a different extensions, as this format differs from `.sierra.json` output of lib compilation?

Reply:
`snforge_sierra.json`?